pkg/commands/kvstoreentry/create.go

```diff
@@ -26,20 +36,63 @@ func NewCreateCommand(parent cmd.Registerer, g *global.Data, m manifest.Data) *C
 		manifest: m,
 	}
 	c.CmdClause = parent.Command("create", "Insert a key-value pair").Alias("insert")
+	c.CmdClause.Flag("key-name", "Key name").Short('k').StringVar(&c.Input.Key)
+	c.CmdClause.Flag("stdin", "Read new-line separated JSON stream via STDIN").BoolVar(&c.stdin)
 	c.CmdClause.Flag("store-id", "Store ID").Short('s').Required().StringVar(&c.Input.ID)
-	c.CmdClause.Flag("key-name", "Key name").Short('k').Required().StringVar(&c.Input.Key)
-	c.CmdClause.Flag("value", "Value").Required().StringVar(&c.Input.Value)
+	c.CmdClause.Flag("value", "Value").StringVar(&c.Input.Value)
 	return &c
 }
 
 // Exec invokes the application logic for the command.
-func (c *CreateCommand) Exec(_ io.Reader, out io.Writer) error {
+func (c *CreateCommand) Exec(in io.Reader, out io.Writer) error {
+	if c.stdin {
+		// Determine if 'in' has data available.
+		if in == nil || text.IsTTY(in) {
+			return fsterr.ErrNoSTDINData
+		}
+
+		host, _ := c.Globals.Endpoint()
+		path := fmt.Sprintf("/resources/stores/kv/%s/batch", c.Input.ID)
+		token, s := c.Globals.Token()
+		if s == lookup.SourceUndefined {
+			return fsterr.ErrNoToken
+		}
+
+		// IMPORTANT: Input could be large so we must buffer the reads.
+		// This will avoid loading all of the data into memory at once.
+		body := bufio.NewReader(in)
+
+		resp, err := undocumented.Call(
+			host, path, http.MethodPut, token, io.TeeReader(body, os.Stdout), c.Globals.HTTPClient,
```

Comment:
I'm only tee'ing the reads so I can see the data get buffered into the HTTP client call.